packages/react-conformance/src/defaultTests.tsx

```diff
@@ -333,6 +336,57 @@ export const defaultTests: TestObject = {
     });
   },
 
+  /** Ensures that components have consistent custom callback names i.e. on[Part][Event] */
+  'consistent-callback-args': (componentInfo, testInfo, tsProgram) => {
+    it('has consistent custom callback arguments (consistent-callback-args)', () => {
+      const { testOptions = {} } = testInfo;
+
+      const propNames = Object.keys(componentInfo.props);
+      const ignoreProps = testOptions['consistent-callback-args']?.ignoreProps || [];
+
+      const invalidProps = propNames.filter(propName => {
+        if (!ignoreProps.includes(propName) && CALLBACK_REGEX.test(propName)) {
+          const propInfo = componentInfo.props[propName];
+
+          if (!propInfo.declarations) {
+            throw new Error(
+              [
+                `Definition for "${propName}" does not have ".declarations".`,
+                'Please report a bug if this happens',
+              ].join(' '),
+            );
+          }
+
+          if (propInfo.declarations.length !== 1) {
+            throw new Error(
+              [
+                `Definition for "${propName}" has multiple elements in ".declarations".`,
+                'Please report a bug if this happens',
+              ].join(' '),
+            );
+          }
+
+          const rootFileName = propInfo.declarations[0].fileName;
+          const typeName = propInfo.declarations[0].name;
+
+          const handlerArguments = getCallbackArguments(tsProgram, rootFileName, typeName, propName);
+
+          if (Object.keys(handlerArguments).length !== 2) {
+            return true;
```

Comment:
It should be possible to assert that both arguments are objects, since you parse the types already ? or at least they aren't primitites like string, number

Reply:
This will be done in a follow up PR as I need to improve resolving logic first (see commented test)